### ISSUE TITLE
openssl s_client is failing with 40D79240AE7F0000:error:8000006F:system library:BIO_connect:Connection refused:/usr/src/debug/openssl/3.1.0-r0/crypto/bio/bio      on Qemu target

### ISSUE THREAD

**Issue by ArchanaWind** [openssl/openssl#20996]
I have built an openssl package myself and booted to qemu target with help of yocto framework
On Qemu emulator I have tried running ```openssl s_client``` command but it is failing with 

40D79240AE7F0000:error:8000006F:system library:BIO_connect:Connection refused:/usr/src/debug/openssl/3.1.0-0/crypto/bio/bio_)
40D79240AE7F0000:error:10000067:BIO routines:BIO_connect:connect error:/usr/src/debug/openssl/3.1.0-0/crypto/bio/bio_sock2.c:
40D79240AE7F0000:error:8000006F:system library:BIO_connect:Connection refused:/usr/src/debug/openssl/3.1.0-0/crypto/bio/bio_)
40D79240AE7F0000:error:10000067:BIO routines:BIO_connect:connect error:/usr/src/debug/openssl/3.1.0-0/crypto/bio/bio_sock2.c:
connect:errno=111

I have tried may ways to resolve but none of them have worked for me
Could any one please help on it

```openssl version```
OpenSSL 3.1.0 14 Mar 2023 (Library: OpenSSL 3.1.0 14 Mar 2023)

Thanks in advance

**Comment by ArchanaWind** [comment]
Hi Tomas Mraz,

Thank you for the quick response,

Yes, It's not OpenSSl issue but I request some help on this
I am trying to run some Nodejs tests on target which are depend on OpenSSL and all tests are failing with "No SSL protocols available" but I am able to list all protocols with below command. so I have raised the issue her to get some help on this
```$openssl ciphers -v | awk '{print $2}' | sort | uniq```
SSLv3
TLSv1
TLSv1.2
TLSv1.3

 

I have enabled network access on VM, Ip address is 192.xx.xx.xx and able to do secure copy files from VM to outside machine and vice versa. but still issue persists.

could you please suggest anything I need to try

thanks in Advance

**Comment by t8m** [comment]
Now that you enabled network on the VM, what are the exact error messages that you see from `openssl s_client`?

**Comment by ArchanaWind** [comment]
The above log attached after enable Netwrork

```# openssl s_client``` 

40477494887F0000:error:8000006F:system library:BIO_connect:Connection refused:/usr/src/debug/openssl/3.1.0-r0/crypto/bio/bio_)
40477494887F0000:error:10000067:BIO routines:BIO_connect:connect error:/usr/src/debug/openssl/3.1.0-r0/crypto/bio/bio_sock2.c:
40477494887F0000:error:8000006F:system library:BIO_connect:Connection refused:/usr/src/debug/openssl/3.1.0-r0/crypto/bio/bio_)
40477494887F0000:error:10000067:BIO routines:BIO_connect:connect error:/usr/src/debug/openssl/3.1.0-r0/crypto/bio/bio_sock2.c:
connect:errno=111

**Comment by t8m** [comment]
Unfortunately `Connection refused` is system error so there is really nothing we can help with here. Are you sure you're connecting to right address/port where a TLS server is listening?

**Comment by ArchanaWind** [comment]
Ok Thank you,

I am connecting right address and port 

Actually I am trying to run Nodejs tests on Qemu emulator, I have built Nodejs with Yocto framework.
The built Nodejs will use system defined OpenSSL. and I am able to run all tests on emulator except TLS related tests 
All TLS tests are failing with ERR_SSL_NO_PROTOCOLS_AVAILABLE

below is the error log

=== release test-tls-getprotocol ===
Path: parallel/test-tls-getprotocol
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: C077EF315B7F0000:error:0A0000BF:SSL routines:tls_setup_handshake:no protocols available:/usr/src/debug/openssl/3.1.0-r0/ssl/statem/statem_lib.c:104:

   ``` at TLSSocket._start (node:_tls_wrap:980:16)
    at Object.onceWrapper (node:events:627:28)
    at TLSSocket.emit (node:events:525:35)
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1479:10)
Emitted 'error' event on TLSSocket instance at:
    at emitErrorNT (node:internal/streams/destroy:151:8)
    at emitErrorCloseNT (node:internal/streams/destroy:116:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  library: 'SSL routines',
  reason: 'no protocols available',
  code: 'ERR_SSL_NO_PROTOCOLS_AVAILABLE'
}
```
Node.js v18.14.2
Command: /usr/bin/node /usr/lib64/nodejs/ptest/test/parallel/test-tls-getprotocol.js

Though it is Nodejs tests but it depends on OpenSSl, so I have requested help here

any help can you provide on this

**Comment by t8m** [comment]
How did you build OpenSSL? Any options passed to the Configure command? How is the OpenSSL invoked from Node.js?  Especially are there any calls to SSL_CTX_set_min/max_proto_version()?

**Comment by nhorman** [comment]
marking as inactive, will be closed at the completion of 3.4 dev without further input
